### PR TITLE
Feat: support apiserver-related multicluster features

### PR DESCRIPTION
**Pull request by Somefive**
#### Feature: Support Add/Remove/Edit Env Placement
This feature is implemented by tracking cluster connections in EnvBindingStatus while making PlacementDecisions. The Garbage Collector will look for cluster connections and run `dispatchAndGC` in those clusters. 

*Notice that when one cluster is unused for one application, all resources will be reclaimed except the ResourceTracker and ControllerRevision. They are kept for future version rollback.*

#### Feature: Disable Trait
This feature allows user to disable trait usage while writing patches in EnvBinding.

#### Feature: Default workflow for EnvBinding
This feature allows user to use EnvBinding even without explicit workflow. It is implemented by automatically generate default workflow when EnvBinding policy exists and workflow not exists, similar to `ApplyComponent`. The implementation also abstracts the WorkflowStepGenerator for future extension.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

This PR is based on #2556.

### CODE REVIEW

**Review by barnettZQG — APPROVED** [review]
LGTM

File: test/e2e-multicluster-test/testdata/app/example-envbinding-app-wo-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: example-app
+  namespace: TEST_NAMESPACE
```
Comment (hongchaodeng) [inline test/e2e-multicluster-test/testdata/app/example-envbinding-app-wo-workflow.yaml:5]:
```suggestion
  namespace: TEST_NAMESPACE # to be replaced
```

**Review by hongchaodeng — APPROVED** [review]
lgtm after nit

File: test/e2e-multicluster-test/testdata/app/example-envbinding-app-wo-workflow.yaml (hunk 1 above)
Comment (Somefive) [reply test/e2e-multicluster-test/testdata/app/example-envbinding-app-wo-workflow.yaml]: ok

**Review by FogDong — APPROVED** [review]

**Review by zzxwill — APPROVED** [review]

**Pull request merged**